### ISSUE TITLE
[TASK] Extract unrolling logic out of Aggregation into Unroll & make it mask-aware

### ISSUE THREAD

**Issue by marcromeyn** [NVIDIA-Merlin/models#713]
In a session-based recsys setting, you often have contextual-features (1-D features like time-of-day, device-type, etc.) & session features (2-D features, a feature per timestep of the sequence). We typically merge these features by unrolling the 1-D features into the shape of the session-features:

![CleanShot 2022-09-05 at 10 42 08](https://user-images.githubusercontent.com/3015785/188407190-e9ffaf3e-1edd-4b35-80ba-a1a736d3d2ac.png)

This logic is now part of the aggregations ([here](https://github.com/NVIDIA-Merlin/models/blob/5a345d9c19fe15cca48bb7a187ff9eb96cc2db84/merlin/models/tf/core/tabular.py#L29-L39) to be specific). We would like to break this out into a new class (working name `Unroll`). 

One reason for this is that we want to make use of the standard way in Keras to propagate masks (more info can be found [here](https://keras.io/guides/understanding_masking_and_padding/)). This new class should implemtent `compute_mask` in order to alter the masks (it should add the mask info for all the contextual features).
